src/internal/RequestFlow.ts

```diff
@@ -56,6 +56,36 @@ export class RequestFlow {
         this.onErrorHandlers.push(handler);
     }
 
+    async execute(interpreter: AquamarineInterpreter, connection: FluenceConnection) {
+        if (this.hasExpired()) {
+            return;
+        }
+
+        logParticle(log.debug, 'interpreter executing particle', this.getParticle());
+        const interpreterOutcome = this.runInterpreter(interpreter);
+
+        log.debug('inner interpreter outcome:', {
+            ret_code: interpreterOutcome.ret_code,
+            error_message: interpreterOutcome.error_message,
+            next_peer_pks: interpreterOutcome.next_peer_pks,
+        });
+
+        if (interpreterOutcome.ret_code !== 0) {
+            this.raiseError(
+                `Interpreter failed with code=${interpreterOutcome.ret_code} message=${interpreterOutcome.error_message}`,
+            );
+        }
+
+        // do nothing if there is no `next_peer_pks` or if client isn't connected to the network
+        if (interpreterOutcome.next_peer_pks.length > 0) {
```

Comment:
Could you please check that `next_peer_pks` contains only relay's PeerId, and throw/log an error otherwise?

Reply:
sounds like a good idea!